woocommerce/wc-admin
# Expand collapse extendable task list

Fixes #6636 

Makes use of the new experimental CollapsibleList from the components package, only using it for the extended task list.

I pointed this PR at `add/6636` as it depends on that PR.

### Accessibility




### Screenshots

The expansion seemed less smooth during the screen recording.
![task-list-expand-collapse](https://user-images.githubusercontent.com/2240960/116588252-b27c3000-a8f1-11eb-96eb-83e10aa58b74.gif)

### Detailed test instructions:

-  Create a 3PD task list by following the JS portion of [this tutorial](https://developer.woocommerce.com/2021/02/26/use-setup-tasks-to-provide-a-first-class-on-boarding-experience-for-merchants/), you can create a new extension using `npm run example -- --ext=add-task` 
OR
[Load this branch of the wc-admin-test-helper](https://github.com/woocommerce/woocommerce-admin-test-helper/pull/17) go to **Tools > WCA Test Helper > Tasks** and update the task number to 5.
Sample of filter:
```
addFilter(
	'woocommerce_admin_onboarding_task_list',
	'woocommerce_gpf',
	( tasks ) => {
		return [
			...tasks,
			{
				key: 'woocommerce_test_item_1',
				title: 'Title',
				container: <TestComp />,
				completed: true,
				visible: true,
				additionalInfo: 'additional info',
				time: '5 minutes',
				isDismissable: true,
			}
                 ]
    }
);
```
- Make sure the above plugin is enabled, and you have added more then 2 items for the `woocommerce_admin_onboarding_task_list` filter.
- Navigate to **WooCommerce > Home**
- Notice how the only 2 items are shown, and you have to expand to see the others
- When expanding and collapsing the list it should trigger these tracks: `wcadmin_extended_tasklist_expand`, `wcadmin_extended_tasklist_collapse`
- Edit you external plugin to only add 2 items
- Navigate back to **WooCommerce > Home** it should not show the expansion item, but only the 2 task items.
